catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/validation/impl/MatchAnyValidatorTest.java

```diff
@@ -48,26 +50,53 @@
 
   private static final Set<String> VALID_ENUMERATIONS_2 = ImmutableSet.of("ben");
 
+  private static final Set<String> VALID_COUNTRY_CODE_ENUMERATIONS =
+      ImmutableSet.of("USA", "CHN", "IND", "ITA", "VNM", "KOR");
+
+  private static final AttributeImpl COUNTRY_CODE_ATTRIBUTE =
+      new AttributeImpl(
+          Location.COUNTRY_CODE, Arrays.asList("USA", "VNM", "KOR", "CHN", "IND", "XKX", "ITA"));
+
+  private static final String COUNTRY_CODE_PATTERN =
+      "^(AA[A-Z])|(Q[M-Z][A-Z])|(X[A-Z][A-Z])|(ZZ[A-Z])$";
+
   private MatchAnyValidator matchAnyValidator;
 
+  private PatternValidator patternValidator;
+
   private EnumerationValidator stubEmptyValidator;
 
   private EnumerationValidator enumerationValidator;
 
   private EnumerationValidator enumerationValidator2;
 
+  private EnumerationValidator countryCodeEnumerationValidator;
+
   @Before
   public void setUp() {
     stubEmptyValidator = mock(EnumerationValidator.class);
     when(stubEmptyValidator.validate(Matchers.any(AttributeImpl.class)))
         .thenReturn(Optional.of(new AttributeValidationReportImpl()));
 
+    patternValidator = new PatternValidator(COUNTRY_CODE_PATTERN);
     enumerationValidator = new EnumerationValidator(VALID_ENUMERATIONS, false);
     enumerationValidator2 = new EnumerationValidator(VALID_ENUMERATIONS_2, false);
+    countryCodeEnumerationValidator =
+        new EnumerationValidator(VALID_COUNTRY_CODE_ENUMERATIONS, false);
+
     SizeValidator sizeValidator = new SizeValidator(1, 3);
     matchAnyValidator = new MatchAnyValidator(Arrays.asList(enumerationValidator, sizeValidator));
   }
 
+  @Test
+  public void testMatchAnyValidatorPassesWhenAttributeIsMultivalued() {
```

Comment:
might be worth adding a test to verify you get a validation report if one of a list of attribute values is invalid.

Reply:
added additional test